single-kernel/flowmap.cpp

```diff
@@ -0,0 +1,211 @@
+#include <iostream>
+#include <sycl/sycl.hpp>
+
+#include "common.h"
+
+namespace s = sycl;
+
+
+class FlowMapKernel; // kernel forward declaration
+
+
+class FlowMap {
+protected:
```

Comment:
Why `protected`? Does anything inherit from this class and access these members? Otherwise this is a bit confusing.

(This applies to multiple benchmarks)